src/ami-set.js

```diff
@@ -8,16 +8,37 @@ const KEY_CONST = 'ami-set';
  * by its AWS region.
  */
 
-let amiSet = base.Entity.configure({
+let AmiSet = base.Entity.configure({
   version: 1,
 
   partitionKey: base.Entity.keys.ConstantKey(KEY_CONST),
-  rowKey: base.Entity.keys.StringKey('amiSetId'),
+  rowKey: base.Entity.keys.StringKey('id'),
 
   properties: {
 
-    amiSetId: base.Entity.types.String,
+    id: base.Entity.types.String,
+    /* This is a JSON object which contains the AMIs of an AMI set keyed by
+     * their virtualization type and region. It is in the shape:
+     * {
```

Comment:
This is not specific to your patch, but I'm sort of wondering if a structure like
```json
{
   "us-west-1": {
      "hvm": "ami-111",
      "pv": "ami-222"
  },
   "us-east-1": {
      "hvm": "ami-111",
      "pv": "ami-222"
  }
}
```
or
```json
[
   {
      "region": "us-west-1",
      "hvm": "ami-111",
      "pv": "ami-222"
   },
   {
      "region": "us-east-1",
      "hvm": "ami-111",
      "pv": "ami-222"
   }
]
```
The reason for flipping the hvm/pv around is that the structure goes from most general (region) to more specific (virtualization style) to most specific (ami id).

Whether we have a list of objects or a straigh mapping is something that you'll have to pick.  A list of objects is really nice because you can do really neat things with `Array.prototype.map` and `Array.prototype.filter`.  A mapping is really nice because it's easier to address the values and it is impossible to have duplicates.  Error checking with the object will be easier, but the list of objects is more in line with what we do elsewhere in the provisioner and probably what I'd prefer to see.